marketplace/netlify-build/src/netlify-app-config.js

```diff
@@ -0,0 +1,254 @@
+      sdk.space.getContentTypes()
+    ]);
+
+    const config = parametersToConfig(parameters);
+    const enabledContentTypes = currentStateToEnabledContentTypes(currentState);
+
+    // First empty site (so no UI click is needed).
+    if (!Array.isArray(config.sites) || config.sites.length < 1) {
+      config.sites = [{}];
+    }
+
+    // Initially we are not connected to Netlify and we don't have
+    // a full list of Netlify sites.
+    // Here we are computing a list consisting of sites we know
+    // of so we can offer Netlify site labels even before connecting.
+    const netlifySites = config.sites
+      .filter(s => s.netlifySiteName)
+      .map(s => ({ id: s.netlifySiteId, name: s.netlifySiteName }));
+
+    const ticketId = await NetlifyClient.createTicket();
+
+    this.setState({
+      ready: true,
+      config,
+      enabledContentTypes,
+      contentTypes: contentTypesResponse.items.map(ct => [ct.sys.id, ct.name]),
+      netlifySites: uniqBy(netlifySites, s => s.id),
+      ticketId
+    });
+  };
+
+  onAppConfigure = async () => {
+    if (!this.state.token) {
+      this.notifyError('You must be connected to Netlify to configure the app.');
+      return false;
+    }
+
+    try {
+      const isInstalled = await this.props.sdk.platformAlpha.app.isInstalled();
+      const method = isInstalled ? 'update' : 'install';
+      const config = await NetlifyIntegration[method]({
+        config: this.state.config, // eslint-disable-line react/no-access-state-in-setstate
+        accessToken: this.state.token // eslint-disable-line react/no-access-state-in-setstate
+      });
+
+      this.setState({ config });
+
+      return {
+        parameters: configToParameters(config),
+        targetState: enabledContentTypesToTargetState(
+          this.state.contentTypes,
+          this.state.enabledContentTypes
+        )
+      };
+    } catch (err) {
+      this.notifyError(err, 'Failed to configure the app.');
+      return false;
+    }
+  };
+
+  notifyError = (err, fallbackMessage) => {
+    let message = fallbackMessage || 'Operation failed.';
+    if (typeof err === 'string') {
+      message = err;
+    } else if (err.useMessage && err.message) {
+      message = err.message;
+    }
+
+    this.props.sdk.notifier.error(message);
+  };
+
+  stopPolling = () => {
+    // Once authentication process is started, we poll for results.
+    // When we leave this page or start a new OAuth flow authentication
+    // process is aborted and the polling needs to be cancelled.
+    if (this.cancelTicketPolling) {
+      this.cancelTicketPolling();
+    }
+  };
+
+  onConnectClick = async () => {
+    this.stopPolling();
+
+    this.cancelTicketPolling = NetlifyClient.getAccessTokenWithTicket(
+      this.state.ticketId,
+      (err, token) => {
+        if (err) {
+          this.notifyError(err, 'Failed to connect with Netlify. Try again!');
+        } else if (token) {
+          this.initNetlifyConnection(token);
+        }
```

Comment:
What happens in the `else` case here where there is no `err` or `token`? Should we alert the user something is wrong?

Reply:
This is handled now.